### DISCUSSION TITLE
fitting momentuHMM with uncertain GPS location

### DISCUSSION THREAD

**Discussion by virginiaIorio** [bmcclintock/momentuHMM#81] [Q&A] [answered]
I am working with telemetry data with GPS tags, which have very little location error, however for certain tracks I have very big time gaps (a few hours) between locations. During these big gaps dive start and end locations are interpolated between GPS locations.
It was suggested to me that to remove this uncertain locations from the model I could have introduced NAs for the step length and turning angle for these locations before fitting the model. For example, if an interpolated location didn’t have a real GPS location within a time window the step length and turning angle of that location would become NAs. This has helped a lot my models to classify different states, especially removing sections of the trips that resulted in straight lines due to dive location interpolation. I was wondering how exactly does the momentuHMM treats and takes into account these NAs?

**Comment by bmcclintock** [comment, marked as answer]
When there are missing locations, the two most common options are: 1) treat the step lengths and turn angles as missing using NAs; or 2) interpolate the missing locations. When missing locations are treated as NAs, both moveHMM and momentuHMM allow state transitions to occur over the missing intervals, but the missing time steps make no contribution to the step length or turning angle components of the likelihood (e.g. see "Estimation, decoding, model selection, and model checking" in [Langrock et al. 2012](https://doi.org/10.1890/11-2241.1)). 

The "best" approach is invariably case-dependent, but simple linear interpolation (SLI) is often OK for filling in very short (and infrequent) gaps, but SLI is generally not a good idea for larger gaps or with >2 movement behavior states. This is because SLI will simply fill in the gaps with straight lines, and the circular distributions commonly used for turning angles can run into problems when there are lots of straight lines in the data. For example, if there are enough of these straight lines, the HMM can have a tendency to assign the gaps to an entirely spurious separate state (with extreme directional persistence) and the non-missing periods to the other state(s).

An alternative to SLI is to use a more "realistic" interpolation method and/or account for the location uncertainty during the gaps. This is what the `crawlWrap` and `MIfitHMM` functions are for.
